xgensum

```diff
@@ -1,21 +1,22 @@
 #!/bin/bash
 # xgensum [-f] [-c] [-i] [-H hostdir] TEMPLATE - update SHA256 sums in templates
 
-case "$1" in
-	-f) FLAG_f=$1; shift
-esac
-
-case "$1" in
-	-c) FLAG_c=$1; shift
-esac
-
-case "$1" in
-	-i*) FLAG_i=$1; shift
-esac
+usage() {
+	echo 'Usage: xgensum [-f] [-c] [-i] [-H hostdir] TEMPLATE' 1>&2
+	exit 1
+}
+
+while getopts fciH:h flag; do
+	case $flag in
+		f) FLAG_f=1 ;;
+		c) FLAG_c=1 ;;
+		i) FLAG_i='-i' ;;
+		H) FLAG_h="-H $OPTARG" ;;
+		h|?) usage ;;
```

Comment:
The ? needs quoting.

Reply:
from how I read the [posix](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/getopts.html) and [bash](https://www.gnu.org/software/bash/manual/html_node/Bourne-Shell-Builtins.html) manuals, doing it without quotes is fine. the posix example even does `?)`

Reply:
and in other xtools it's unquoted (albeit one is from me):

https://github.com/leahneukirchen/xtools/blob/8a37db0bb271efb8948c615beef8ff7583109ed5/xpkg#L10-L33

https://github.com/leahneukirchen/xtools/blob/dc3688eb4fa1e475dd0ac6906fc16a261d46fafc/xpkgdiff#L16-L27

Reply:
Yes, sorry, this was a brainfart.